### PR TITLE
[Merged by Bors] - feat(Logic/Basic): xor_iff_or_and_not_and

### PR DESCRIPTION
**Pull request by mans0954**
Characterises `Xor'` as "one or the other but not both".

c.f. `symmDiff_eq'` in `Order/SymmDiff`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Basic.lean [hunk 1]
```diff
@@ -253,6 +253,14 @@ instance : Std.Commutative Xor' := ⟨xor_comm⟩
 
 theorem xor_not_not : Xor' (¬a) (¬b) ↔ Xor' a b := by simp [Xor', or_comm, and_comm]
 
+@[simp] theorem and_not_self_and_iff_and_not (a b : Prop) :  a ∧ (a → ¬b) ↔ a ∧ ¬b := by
+  rw [and_congr_right_iff]
+  exact imp_iff_right
+
+theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
+  rw [Xor', or_and_right, not_and, and_not_self_and_iff_and_not, imp_not_comm,
+    and_not_self_and_iff_and_not (b := a) (a := b)]
```
Comment (ADedecker) [inline Mathlib/Logic/Basic.lean:262]:
```suggestion
theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
  simp_rw [Xor', or_and_right, not_and_or, and_or_left, and_not_self_iff, false_or, or_false]
```

Unless you have some other reason to include that other lemma of course.

Comment (mans0954) [reply Mathlib/Logic/Basic.lean]: @ADedecker that doesn't appear to work - or at least depends on tactics / results not yet available at this point in the build.

Comment (mans0954) [reply Mathlib/Logic/Basic.lean]: How about?
```
theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
  rw [Xor', or_and_right, not_and_or, and_or_left, and_not_self_iff, false_or,
    and_or_left, and_not_self_iff, or_false]
```

Comment (ADedecker) [reply Mathlib/Logic/Basic.lean]: TIL `simp_rw` is not imported by all of Mathlib! Sorry for the troubles.

**Pull request closed without merging**